pkg/kapp/cmd/appchange/describe.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package appchange
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/cppforlife/go-cli-ui/ui"
+	ctlapp "github.com/k14s/kapp/pkg/kapp/app"
+	cmdapp "github.com/k14s/kapp/pkg/kapp/cmd/app"
+	cmdcore "github.com/k14s/kapp/pkg/kapp/cmd/core"
+	"github.com/k14s/kapp/pkg/kapp/logger"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type DescribeOptions struct {
+	ui          ui.UI
+	depsFactory cmdcore.DepsFactory
+	logger      logger.Logger
+
+	appChange string
+
+	AppFlags       cmdapp.Flags
+	NamespaceFlags cmdcore.NamespaceFlags
+}
+
+func NewDescribeOptions(ui ui.UI, depsFactory cmdcore.DepsFactory, logger logger.Logger) *DescribeOptions {
+	return &DescribeOptions{ui: ui, depsFactory: depsFactory, logger: logger}
+}
+
+func NewDescribeCmd(o *DescribeOptions, flagsFactory cmdcore.FlagsFactory) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "describe",
+		Aliases: []string{"d"},
+		Short:   "View changes applied for app-change",
+		RunE:    func(_ *cobra.Command, _ []string) error { return o.Run() },
+	}
+	o.AppFlags.Set(cmd, flagsFactory)
+	cmd.Flags().StringVar(&o.appChange, "app-change", "", "Name of app-change to be described")
+	return cmd
+}
+
+func (o *DescribeOptions) Run() error {
+	if o.appChange == "" {
+		return fmt.Errorf("Flag --app-change is required")
+	}
+
+	client, err := o.depsFactory.CoreClient()
+	if err != nil {
+		return err
+	}
+
+	appChangeMap, err := client.CoreV1().ConfigMaps(o.AppFlags.NamespaceFlags.Name).Get(context.TODO(), o.appChange, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	_, found := appChangeMap.Labels[ctlapp.IsChangeLabelKey]
+	if !found {
+		return fmt.Errorf("Not an app-change")
+	}
+
+	var configSpec ctlapp.ChangeMeta
+	err = json.Unmarshal([]byte(appChangeMap.Data[ctlapp.ChangeMetaKey]), &configSpec)
+	if err != nil {
+		return fmt.Errorf("Unmarshalling app-change spec: %s", err.Error())
+	}
```

Comment:
we have following interface -- https://github.com/vmware-tanzu/carvel-kapp/blob/develop/pkg/kapp/app/interfaces.go#L26-L30. whats the alternative way of implementing above.

Reply:
Oh I see, it will more sense to have a function added to the interface than to make the private variables public for this implementation. Will do!

Reply:
We could query a list of Changes through the interface and then grab the appropriate Meta instead of querying with the core client directly. I see that the app label and app-change label are already being added as selectors there 🤔

Reply:
Will play around with that approach

Reply:
As of now the user is not asked to specify the app itself, but this approach will require that